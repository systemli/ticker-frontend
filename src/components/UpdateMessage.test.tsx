import React from 'react'
import { render, screen } from '@testing-library/react'
import { UpdateMessage } from './index'

describe('UpdateMessage', function () {
  const original = window.location

  beforeAll(() => {
    Object.defineProperty(window, 'location', {
      configurable: true,
      value: { reload: jest.fn() },
    })
  })

  afterAll(() => {
    Object.defineProperty(window, 'location', {
      configurable: true,
      value: original,
    })
  })

  test('renders correctly when active', function () {
    const { asFragment } = render(<UpdateMessage update />)

    expect(asFragment()).toMatchSnapshot()
  })

  test('renders nothing when dismissed', function () {
    const { asFragment } = render(<UpdateMessage update={false} />)

    expect(asFragment()).toMatchSnapshot()
  })

  test('triggers reload correctly', function () {
    render(<UpdateMessage update />)

    expect(
      screen.findByText('An update is available. Click here to update the App.')
    )

    screen.getByText('here').click()

    expect(window.location.reload).toHaveBeenCalled()
  })
})
