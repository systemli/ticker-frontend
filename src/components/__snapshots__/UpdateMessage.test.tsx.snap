// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateMessage renders correctly when active 1`] = `
<DocumentFragment>
  <div
    class="sc-iBkjds hnOzZn"
  >
    <div
      class="ui yellow negative message"
    >
      An update is available. Click 
      <a
        class="sc-ftvSup uBGUH"
      >
        here
      </a>
       to update the App.
    </div>
  </div>
</DocumentFragment>
`;

exports[`UpdateMessage renders nothing when dismissed 1`] = `<DocumentFragment />`;
